ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/StatsUpdater.java

```diff
@@ -73,6 +72,18 @@ public void gatherStats(CompactionInfo ci, HiveConf conf, String userName, Strin
                 sb.append(")");
             }
             sb.append(" compute statistics");
+            if (!conf.getBoolVar(HiveConf.ConfVars.HIVESTATSAUTOGATHER) && ci.isMajorCompaction() && msc != null) {
```

Comment:
check msc at the beginning of the method and throw an exception if null.

Reply:
Done.